VisualPinball.Unity/VisualPinball.Unity.Editor/Managers/ManagerWindow.cs

```diff
@@ -0,0 +1,288 @@
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEditor;
+using UnityEditor.IMGUI.Controls;
+using UnityEngine;
+using VisualPinball.Unity.Extensions;
+using VisualPinball.Unity.VPT;
+using VisualPinball.Unity.VPT.Table;
+
+namespace VisualPinball.Unity.Editor.Managers
+{
+	/// <summary>
+	/// Base class for VPX-style "Manager" windows, such as the Material Manager
+	/// </summary>
+	/// <typeparam name="T">class of type IManagerListData that represents the data being edited</typeparam>
+	public abstract class ManagerWindow<T> : EditorWindow where T: class, IManagerListData
+	{
+		protected virtual string DataTypeName => "";
+
+		protected virtual void OnButtonBarGUI() { }
+		protected virtual void OnDataDetailGUI() { }
+		protected virtual void RenameExistingItem(T data, string desiredName) { }
+		protected virtual void CollectData(List<T> data) { }
```

Comment:
Would be more comfortable to understand if it was `protected virtual List<T> CollectData(List<T> data) => new List<T>()`, no?

Reply:
Yes absolutely